src/modules/hardware-wallet/src/HardwareWallet.ts

```diff
@@ -0,0 +1,80 @@
+import { Instruction, Device, ConnEvent } from './types'
+import { Subscription } from 'rxjs'
+
+const isNodeEnvironment = typeof module !== 'undefined' && this.module !== module
+
+let transportNodeHid
+let outResolve
+
+const isImported = new Promise(async (resolve, reject) => {
+    outResolve = resolve
+})
+
```

Comment:
This is a little hacky but necessary (imo) way to import the hardware transport module only if in a NodeJS environment.

If we want to support it in the browser in the future, we would need to import a different library in that case. But for now, only node is supported.

Also, since doing an import in this particular way is async, I'm awaiting a promise in the other methods that depend on that library. After it is imported, I resolve it.